spec/requests/api/v1/targets/create_spec.rb

```diff
@@ -78,4 +78,28 @@
       expect { subject }.not_to change { Target.count }
     end
   end
+
+  context 'when the user is not authenticated' do
+    subject { post api_v1_targets_url, params: target_params, as: :json }
+
+    let(:target_params) do
+      {
+        target: attributes_for(:target).merge(topic_id: topic.id)
+      }
+    end
```

Comment:
I think it will work the same way If you just do 
`let(:target_params) { nil } `
As params does not get to be checked, becouse authentication erros comes first.

Reply:
Thanks, you are right authentications error come first.